lib/stat_tracker.rb

```diff
@@ -21,6 +21,60 @@ def initialize(files)
     end
   end
 
+  def highest_scoring_visitor 
+    #make a hash of of total away games by team id
+    away_games_by_team_id = @games.each_with_object(Hash.new(0.0)) do |game, hash|
+      hash[game.away_team_id] += 1
+    end
+    #make a hash of total away goals by team id
+    away_goals_by_team_id = @games.each_with_object(Hash.new(0.0)) do |game, hash|
+      hash[game.away_team_id] += game.away_goals
```

Comment:
Great job at chunking and breaking down problems into base levels. For a future refactor, we can iterate @games once and set the hash value to an array to store both game count and total goals. This will increase space and time complexity.

Reply:
for sure! I was trying to keep it simple, but definitely see the value in that since we're running through some large files. Will implement those changes and essentially combine the averaging as an array in the key value pair with the key being the team id.